File: 392.判断子序列/isSubsequence.go
package problem0392

// 给定字符串s和t，判断s是否为t的子序列
// 双指针解法
func isSubsequence(s string, t string) bool {
	if len(s) == 0 {
		return true
	}
	j := 0
	for i := 0; i < len(t); i++ {
		if s[j] == t[i] {
			j++
			if j == len(s) {
				return true
			}
		}

	}
	return false
}

File: 392.判断子序列/isSubsequence.js
// 显然是双指针的解法
function isSubsequence(s, t) {
  if (s.length > t.length) {
    return false
  }
  const sLen = s.length
  const tLen = t.length
  let i = 0
  let j = 0
  while (i < sLen) {
    while (s[i] !== t[j]) {
      j++
      if (j >= tLen) {
        return false
      }
    }
    i++
    j++
  }
  return true
}
